src/vic3/data/scripted_rules.rs

```diff
@@ -0,0 +1,125 @@
+use fnv::FnvHashMap;
+use once_cell::sync::Lazy;
+
+use crate::block::Block;
+use crate::context::ScopeContext;
+use crate::db::{Db, DbKind};
+use crate::everything::Everything;
+use crate::game::GameFlags;
+use crate::item::{Item, ItemLoader};
+use crate::parse::pdxfile::parse_pdx_internal;
+use crate::report::{err, ErrorKey};
+use crate::scopes::Scopes;
+use crate::token::Token;
+use crate::tooltipped::Tooltipped;
+use crate::trigger::validate_trigger;
+
+#[derive(Clone, Debug)]
+pub struct ScriptedRule {}
+
+inventory::submit! {
+    ItemLoader::Normal(GameFlags::Vic3, Item::ScriptedRule, ScriptedRule::add)
+}
+
+impl ScriptedRule {
+    pub fn add(db: &mut Db, key: Token, block: Block) {
+        db.add(Item::ScriptedRule, key, block, Box::new(Self {}));
+    }
+}
+
+impl DbKind for ScriptedRule {
+    fn validate(&self, key: &Token, block: &Block, data: &Everything) {
+        if let Some(sr_sc) = SCRIPTED_RULE_SCOPES_MAP.get(key.as_str()) {
+            let mut sc = ScopeContext::new(sr_sc.root, key);
+            for (name, s) in &sr_sc.names {
+                sc.define_name(name, *s, key);
+            }
+            for (list, s) in &sr_sc.names {
+                sc.define_list(list, *s, key);
+            }
+            validate_trigger(block, data, &mut sc, sr_sc.tooltipped);
+        } else {
+            let msg = "unknown scripted rule";
+            err(ErrorKey::Validation).msg(msg).loc(key).push();
+            let mut sc = ScopeContext::new(Scopes::non_primitive(), key);
+            sc.set_strict_scopes(false);
+            validate_trigger(block, data, &mut sc, Tooltipped::No);
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+struct ScriptedRuleScopeContext {
+    tooltipped: Tooltipped,
+    root: Scopes,
+    names: Vec<(String, Scopes)>,
+    lists: Vec<(String, Scopes)>,
+}
+
+/// Processed version of [`SCRIPTED_RULES`].
+static SCRIPTED_RULE_SCOPES_MAP: Lazy<FnvHashMap<String, ScriptedRuleScopeContext>> =
```

Comment:
Would it be better to simply save them as `ScriptedRuleScopeContext`s rather than being parsed?

Reply:
The parsing code was copied from ck3 :) I think we'll be glad of the flexibility if more rules are added

Reply:
Unifying the parsing machinery might be nice but I'm not sure how